### ISSUE TITLE
Error rendering in core.js

### ISSUE THREAD

**Issue by joelgriffith** [conorhastings/react-syntax-highlighter#83]
Love the library, I agree with all the underlying principals! Having issues (at least in the lite version):

```
Uncaught (in promise) Error: Expected `string` for value, got `var a = "wat";`
```

I think this is due to `children` being an array and being passed directly through? Here's my usage:

```
import SyntaxHighlighter, { registerLanguage } from 'react-syntax-highlighter/dist/light';
import js from 'react-syntax-highlighter/dist/languages/javascript';
import docco from 'react-syntax-highlighter/dist/styles/docco';

registerLanguage('javascript', js);

// render() =>
<SyntaxHighlighter language="javascript" style={docco}>
{ 'var a = "wat";' }
</SyntaxHighlighter>
```

**Comment by conorhastings** [comment]
hmm, I can't seem to reproduce - https://www.webpackbin.com/bins/-KtPA9bNnvUmcroaCU5B

do you have any more info perhaps?

**Comment by adack123** [comment]
@conorhastings I'm also experiencing something very similar to @joelgriffith. Even with just a bare bones example I'm still getting that error

`Uncaught (in promise) Error: Expected `string` for value, ...`

```javascript
const Test = () => (
  <SyntaxHighlighter language="javascript" style={docco}>
    {'var a = "wat";'}
  </SyntaxHighlighter>
);
```
Any ideas?

**Comment by joelgriffith** [comment]
I've found the offending line here:

```js
// lowlight/lib/core.js
function coreHighlight(name, value, ignore, prefix, continuation) {
  var continuations = {};
  var stack = [];
  var modeBuffer = EMPTY;
  var relevance = 0;
  var language;
  var top;
  var current;
  var currentChildren;
  var offset;
  var count;
  var match;
  var children;

  if (typeof name !== 'string') {
    throw new Error('Expected `string` for name, got `' + name + '`');
  }
  // Happens here as children props is proxied through (and is an array)
  if (typeof value !== 'string') {
    throw new Error('Expected `string` for value, got `' + value + '`');
  }
```

A few other things to note: I'm using `preact` with their compat library. I don't think that should effect the bug that much as react itself also exposes `children` as an array as well. Just wanted to makes sure that was understood.

**Comment by conorhastings** [comment]
@joelgriffith I'm relatively certain that react passes children as whatever is passed into React.createElement while im not sure what preact does

**Comment by goto1** [comment]
also having the same exact issue... `Expected `string` for value, got `(num) => num + 1`` with the example `light build` example ... using a `preact` project that was generated using `preact-cli`, but it supposedly is compatible with `react` libraries, so it shouldn't be much of an issue...

**Comment by joelgriffith** [comment]
I'll do some investigation today. Definitely feels like a preact or preact-compat issue. I'll report back to this thread once I hunt it down

**Comment by conorhastings** [comment]
@joelgriffith did you ever find anything?

**Comment by joelgriffith** [comment]
Unfortunately not, I think this is has to do with a slight implementation difference between preact and react... of which I've found a few. I'll see if I can dive deeper and find an answer

**Comment by conorhastings** [comment]
@joelgriffith I'm relatively familiar with preact so I'd love to find a solution, is there anyway you can put together a reproduce use case on https://webpackbin.com that I can take a look at it?

**Comment by joelgriffith** [comment]
Hrm, webpackbin doesn't seem to support resolve aliases, which is how this is supposed to be setup with preact.

Here's a repo instead: https://github.com/joelgriffith/react-syntax-preact-bug. In particular is https://github.com/joelgriffith/react-syntax-preact-bug/blob/master/src/Hello.js where I mount and use the highlighter

**Comment by conorhastings** [comment]
🆒 ! thanks, I'll check it out tonight

**Comment by joelgriffith** [comment]
Thanks Conor! Really appreciate your persistence here, let me know if I can jump in and help

**Comment by conorhastings** [comment]
@joelgriffith believe it's resolved in `v5.7.1` , can you confirm?

**Comment by joelgriffith** [comment]
🌭 ! Fantastic, works great!
